pkg/webhook/storageclass_test.go

```diff
@@ -0,0 +1,155 @@
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestMutateStorageClass(t *testing.T) {
+	storageClassName := "filestore-multishare"
+	labelName := "multishare-label"
+
+	testCases := []struct {
+		name         string
+		storageClass *storagev1.StorageClass
+		operation    v1.Operation
+		shouldAdmit  bool
+		patch        string
+		msg          string
+	}{
+		{
+			name:        "Update should not be allowed",
+			operation:   v1.Update,
+			shouldAdmit: false,
+			msg:         fmt.Errorf("resource %v is immutable", &StorageClassV1GVR).Error(),
+		},
+		{
+			name: "create with non-multishare should be allowed",
+			storageClass: &storagev1.StorageClass{
+				ObjectMeta:  metav1.ObjectMeta{Name: storageClassName},
+				Provisioner: FilestoreCSIDriver,
+			},
+			operation:   v1.Create,
+			shouldAdmit: true,
+		},
+		{
+			name: "create with other provisioner should be allowed",
+			storageClass: &storagev1.StorageClass{
+				ObjectMeta:  metav1.ObjectMeta{Name: storageClassName},
+				Provisioner: "pd.csi.storage.gke.io",
+			},
+			operation:   v1.Create,
+			shouldAdmit: true,
+		},
+		{
+			name: "create with multishare but default tier should not be allowed",
+			storageClass: &storagev1.StorageClass{
+				ObjectMeta:  metav1.ObjectMeta{Name: storageClassName},
+				Provisioner: FilestoreCSIDriver,
+				Parameters: map[string]string{
+					"multishare": "true",
+				},
+			},
+			operation:   v1.Create,
+			shouldAdmit: false,
+			msg:         fmt.Errorf("mutlishare is only supported on %q tier instances", TierEnterprise).Error(),
+		},
+		{
+			name: "create with multishare but not enterprise tier should not be allowed",
+			storageClass: &storagev1.StorageClass{
+				ObjectMeta:  metav1.ObjectMeta{Name: storageClassName},
+				Provisioner: FilestoreCSIDriver,
+				Parameters: map[string]string{
+					"multishare": "true",
+					"tier":       "performance",
+				},
+			},
+			operation:   v1.Create,
+			shouldAdmit: false,
+			msg:         fmt.Errorf("mutlishare is only supported on %q tier instances", TierEnterprise).Error(),
+		},
+		{
+			name: "should fill in instanceStorageClassLabel if not present",
+			storageClass: &storagev1.StorageClass{
+				ObjectMeta:  metav1.ObjectMeta{Name: storageClassName},
+				Provisioner: FilestoreCSIDriver,
+				Parameters: map[string]string{
+					"multishare": "true",
+					"tier":       TierEnterprise,
+				},
+			},
+			operation:   v1.Create,
+			shouldAdmit: true,
+			patch:       fmt.Sprintf(`[{"op":"add", "path":"/parameters/%s","value": "%s"}]`, InstanceStorageClassLabel, storageClassName),
+		},
+		{
+			name: "should not change instanceStorageClassLabel if already set",
+			storageClass: &storagev1.StorageClass{
+				ObjectMeta:  metav1.ObjectMeta{Name: storageClassName},
+				Provisioner: FilestoreCSIDriver,
+				Parameters: map[string]string{
+					"multishare":              "true",
+					"tier":                    TierEnterprise,
+					InstanceStorageClassLabel: labelName,
```

Comment:
add test case for InstanceStorageClassLabel with invalid label values (which dont comply with label value req as pointed above)

Reply:
done